sunpy/map/maputils.py

```diff
@@ -124,6 +124,17 @@ def sample_at_coords(smap, coordinates):
     return smap.data[smap.wcs.world_to_array_index(coordinates)]
 
 
+def _edge_coordinates(smap):
+    # Calculate all the edge pixels
+    edges = map_edges(smap)
+    edge_pixels = list(chain.from_iterable([edges[0], edges[1], edges[2], edges[3]]))
+    x = [p[0] for p in edge_pixels] * u.pix
+    y = [p[1] for p in edge_pixels] * u.pix
+
+    # Calculate the edge of the world
+    return smap.pixel_to_world(x, y)
```

Comment:
Having said that wouldn't casting to a numpy array be a better idea?

```suggestion
    edge_pixels = list(chain.from_iterable([edges[0], edges[1], edges[2], edges[3]])) * u.pix
    return smap.pixel_to_world(*edge_pixels.T)
```

Reply:
Done something slightly different, but shaved around 50% of the new times 🕐